### PR TITLE
Handling network errors

### PR DESCRIPTION
**Pull request by kevingilliard**
- Add retries with exponential backoff
- Handle HTTP Response codes are appropriately.

### CODE REVIEW

File: Analytics/Request/BlockingRequestHandler.cs [hunk 1]
```diff
@@ -131,41 +132,88 @@ internal BlockingRequestHandler(Client client, TimeSpan timeout)
...
+					}
+					catch (WebException ex)
+					{
+						watch.Stop();
+
+						var response = (HttpWebResponse)ex.Response;
+						if (response != null)
+						{
+							statusCode = (int)response.StatusCode;
+							if (statusCode >= 500 || statusCode == 429)
```
Comment (f2prateek) [inline Analytics/Request/BlockingRequestHandler.cs]: Let's make this condition: `if ((statusCode >= 500 && statusCode <= 600) || statusCode == 429)`. Technically there's a HTTP 6xx class of codes.

File: Analytics/Request/BlockingRequestHandler.cs [hunk 2]
```diff
@@ -131,41 +132,88 @@ internal BlockingRequestHandler(Client client, TimeSpan timeout)
 					{ "batch size", batch.batch.Count }
 				});
 
+				const int MAXIMUM_BACKOFF_DURATION = 10000;
```
Comment (f2prateek) [inline Analytics/Request/BlockingRequestHandler.cs]: Can you add a comment indicating this is 10seconds? `const int MAXIMUM_BACKOFF_DURATION = 10000; // 10 seconds`.

File: Analytics/Request/BlockingRequestHandler.cs [hunk 3]
```diff
@@ -131,41 +132,88 @@ internal BlockingRequestHandler(Client client, TimeSpan timeout)
...
+				int backoff = 100;
```
Comment (f2prateek) [inline Analytics/Request/BlockingRequestHandler.cs]: Same as above, let's add a comment explaining this is 100 ms.

**Comment by kevingilliard** [comment]
I've changed the files as you commented.

**Pull request merged**